src/API/Endpoints/Reports/Income.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * Income over time endpoint
+ *
+ * @package Give
+ */
+
+namespace Give\API\Endpoints\Reports;
+
+class Income extends Endpoint {
+
+	public function __construct() {
+		$this->endpoint = 'income';
+	}
+
+	public function get_report( $request ) {
+
+		// Check if a cached version exists
+		$cached_report = $this->get_cached_report( $request );
+		if ( $cached_report !== false ) {
+			// Bail and return the cached version
+			return new \WP_REST_Response(
+				[
+					'data' => $cached_report,
+				]
+			);
+		}
+
+		$start = date_create( $request['start'] );
+		$end   = date_create( $request['end'] );
+		$diff  = date_diff( $start, $end );
+
+		$data = [];
+
+		switch ( true ) {
```

Comment:
What did you find with regards to the time axis automatically handling the ticks for you?

Reply:
I did some experimentation to that end, but quickly realized it requires some fairly size-able rethinking of the Chart component, so I'm opening a separate issue to consider this and sticking to what works for now.

Reply:
Roger that!